File: src/db/schema/product.ts
import { sqliteTable, text, integer, real } from "drizzle-orm/sqlite-core";
import { v4 as uuidv4 } from "uuid";

export const productSqlite = sqliteTable("products", {
  id: text("id").primaryKey().$defaultFn(() => uuidv4()),

  name: text("name").notNull(),
  code: text("code"),
  description: text("description"),

  categoryId: text("category_id"),

  price: real("price").notNull(),
  active: integer("active").default(1),
  sortOrder: integer("sort_order").default(0),

  createdAt: text("created_at"),
  updatedAt: text("updated_at"),
  deletedAt: text("deleted_at"),

  media: text("media"), 
});

File: src-tauri/src/db/models/product.rs

use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Product {
    pub id: String,
    pub name: String,
    pub code: Option<String>,
    pub description: Option<String>,
    pub category_id: Option<String>,
    pub price: f64,
    pub active: bool,
    pub sort_order: i32,
    pub is_sold_out: Option<i32>,

    pub created_at: Option<String>,
    pub updated_at: Option<String>,
    pub deleted_at: Option<String>,

    // store as JSON string
    pub media: Option<String>,
    pub overrides:Option<String>,
}
